### ISSUE TITLE
Table of Contents: fix the wrong aria-label usage

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#2710]
Latest changes introduced in the Table of Contents added an aria-label attribute to the buttons with the heading titles. 

The heading titles in the TOC are now clickable: clicking on them moves focus to the related heading in the content so users can edit right away. That's good, however it's not so clear headings are actually clickable:
- visually, there's no indication the headings in the TOC are actually clickable buttons
- the focus style needs to be improved (currently, it's just the native browsers outline: on webkit macOS this appears like a blue big outline)
- the `aria-label="Focus heading block"` completely overrides the buttons content, thus making impossible for assistive technology users to get the heading title

Worth reminding [aria-label overrides the UI controls accessible name](https://www.w3.org/TR/wai-aria/states_and_properties#aria-label), thus the only thing screen readers announce is the aria label value "Focus heading block":

![screen shot 2017-09-09 at 14 19 27](https://user-images.githubusercontent.com/1682452/30240086-f9905c90-9569-11e7-8cc4-fe34a37ba4eb.png)

From a UI perspective, user interface controls should be labeled to indicate the available action (e.g. "Edit this", "Set that", etc.) and not with the value or name of the object they refer to. This happens also for other UI controls in Gutenberg and has been mentioned also in other issues (check for example the Visibility and Publish settings in the sidebar.

The pattern currently used in WordPress is to label controls indicating the available action. Gutenberg introduces a new pattern that, as I see it, is far from ideal from both a design/usability and accessibility perspective.

Not to argue, but I think this pattern was just copied from Calypso without a thorough thinking. In the case, for example, of the visibility settings, landing on a control that says just "Public" doesn't tell me so much. After changing the setting, the UI controls label changes, for example, to "Private". Instead, controls label should make sense even when read out of context and always be identified with the same label.

This partially applies also to the headings in the TOC. While the heading titles don't change, using as button labels the headings title doesn't tell users anything about the available action. 
Vice-versa, overriding the label with `aria-label="Focus heading block"` doesn't tell assistive technologies users anything about _which_ heading.

Instead of mixing values and actions, a best option would be providing the two information separately. I'd like to propose two options:
- provide a separate control to move focus, and keep the heading titles as just text
- or, instead of overriding the heading titles, provide the info about the available action as additional info. This pattern is already used in some places in WordPress, for example the Posts list where the post titles use an aria label: `aria-label="“My nice post” (Edit)"`. This wouldn't solve the visual issue though.

Visually, it's still not clear headings are clickable. See the screenshot below: would you understand at first sight they can be clicked?

<img width="313" alt="screen shot 2017-09-09 at 14 57 38" src="https://user-images.githubusercontent.com/1682452/30240365-69ee95ec-956f-11e7-970a-5f0cfc9ad6f9.png">

**Comment by karmatosed** [comment]
I am ok about visually indicating that they are clickable, but adding another strong border box around them like we have an issue to fix in already, I do not feel is the solution.

**Comment by afercia** [comment]
This specific issue has been fixed, now the buttons have a visually hidden text:
`<span class="screen-reader-text">(Click to focus this heading)</span>`

This word count/table of content panel has other accessibility problems (toggle button label, placement, focus management, etc.) but they should be addressed in a separate issue.
